main/src/addins/MonoDevelop.DesignerSupport/MonoDevelop.DesignerSupport/NativePropertyEditor/ComponentModelObjectEditor.cs

```diff
@@ -0,0 +1,142 @@
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using MonoDevelop.Core;
+using Xamarin.PropertyEditing;
+
+namespace MonoDevelop.DesignerSupport
+{
+	class ComponentModelObjectEditor
+		: IObjectEditor, INameableObject
+	{
+		private readonly ComponentModelTarget propertyItem;
+		public string Name { get; private set; }
+
+		static IReadOnlyList<string> defaultHandlerList = new List<string> ().AsReadOnly ();
+		static AssignableTypesResult defaultAssignableTypeResult = new AssignableTypesResult (new List<ITypeInfo> ().AsReadOnly ());
+		private static readonly IObjectEditor [] EmptyDirectChildren = Array.Empty<IObjectEditor> ();
+
+		private readonly List<IPropertyInfo> properties = new List<IPropertyInfo> ();
+		private readonly List<PropertyDescriptorEventInfo> events = new List<PropertyDescriptorEventInfo> ();
+
+		public object Target => this.propertyItem;
+
+		public ITypeInfo TargetType => ToTypeInfo (Target.GetType ());
+
+		public static ITypeInfo ToTypeInfo (Type type, bool isRelevant = true)
+		{
+			var asm = type.Assembly.GetName ().Name;
+			return new TypeInfo (new AssemblyInfo (asm, isRelevant), type.Namespace, type.Name);
+		}
+
+		public IReadOnlyCollection<IPropertyInfo> Properties => this.properties;
+
+		public IReadOnlyDictionary<IPropertyInfo, KnownProperty> KnownProperties => null;
+
+		public IObjectEditor Parent => null;
+
+		public IReadOnlyList<IObjectEditor> DirectChildren => EmptyDirectChildren;
+		public IReadOnlyCollection<IEventInfo> Events => this.events;
+
+		internal static Task<AssignableTypesResult> GetAssignableTypes ()
+			=> Task.FromResult (defaultAssignableTypeResult);
+
+		public event EventHandler<EditorPropertyChangedEventArgs> PropertyChanged;
+
+		public ComponentModelObjectEditor (ComponentModelTarget propertyItem)
+		{
+			this.propertyItem = propertyItem;
+			this.properties.AddRange (ComponentModelEditorProvider.GetPropertiesForProviders (propertyItem.Providers));
+		}
+
+		public async Task<AssignableTypesResult> GetAssignableTypesAsync (IPropertyInfo property, bool childTypes)
+		{
+			return new AssignableTypesResult (Array.Empty<ITypeInfo> ());
+		}
+
+		public Task<IReadOnlyList<string>> GetHandlersAsync (IEventInfo ev) => Task.FromResult(defaultHandlerList);
+
+		public Task<string> GetNameAsync () => Task.FromResult (Name);
+
+		public Task<IReadOnlyCollection<PropertyVariation>> GetPropertyVariantsAsync (IPropertyInfo property)
+		 => Task.FromResult<IReadOnlyCollection<PropertyVariation>> (Array.Empty<PropertyVariation> ());
+
+		public async Task<ValueInfo<T>> GetValueAsync<T> (IPropertyInfo property, PropertyVariation variations = null)
+		{
+			if (property == null)
+				throw new ArgumentNullException (nameof (property));
+
+			if (!(property is DescriptorPropertyInfo propertyInfo))
+				throw new ArgumentException ("Property should be a DescriptorPropertyInfo", nameof (property));
+
+			T value = await propertyInfo.GetValueAsync<T> (this);
```

Comment:
Does threading matter here?

```suggestion
			T value = await propertyInfo.GetValueAsync<T> (this).ConfigureAwait (false);
```

Reply:
Fixed